metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -79,6 +78,8 @@ class KubernetesDecorator(StepDecorator):
         "namespace": None,
         "gpu": None,  # value of 0 implies that the scheduled node should not have GPUs
         "gpu_vendor": None,
+        "tolerations": None,  # e.g., [{"key": "arch", "operator": "Equal", "value": "amd"}],
```

Comment:
@odracci can you verify how tolerations would be passed through CLI - `python flow.py run --with kubernetes:tolerations=blah..`

Reply:
it doesn't work. I think there is a bug in `BaseCommand.invoke`.
I ran `python flow.py run --with 'kubernetes:tolerations=[{"key":"app","effect":"NoSchedule","value":"foo"}]'` and it fails. Logs:
```
2022-12-06 15:54:56.596 Workflow starting (run-id 1030):
2022-12-06 15:54:58.335 [1030/start/73637 (pid 25832)] Task is starting.
2022-12-06 15:55:00.021 [1030/start/73637 (pid 25832)] Usage: workflow_simple.py kubernetes step [OPTIONS] STEP_NAME CODE_PACKAGE_SHA
2022-12-06 15:55:00.300 [1030/start/73637 (pid 25832)] CODE_PACKAGE_URL
2022-12-06 15:55:00.300 [1030/start/73637 (pid 25832)] Try 'workflow_simple.py kubernetes step --help' for help.
2022-12-06 15:55:00.301 [1030/start/73637 (pid 25832)]
2022-12-06 15:55:00.301 [1030/start/73637 (pid 25832)] Error: Invalid value for '--tolerations': {'key': 'app', 'effect': 'NoSchedule', 'value': 'foo'} is not a valid JSON object
2022-12-06 15:55:00.436 [1030/start/73637 (pid 25832)] Task failed.
```

I debugged the script. The main thread has the correct JSON string, then when Metaflow invokes a new process (pid 25832), the parameter becomes a string like `{'key': 'app', 'effect': 'NoSchedule', 'value': 'foo'}`, it replaces double quotes with quotes

Reply:
solved